caikit/runtime/service_generation/signature_parsing/docstrings.py

```diff
@@ -224,17 +208,15 @@ def _get_docstring_type(
                 f"Found valid candidate type on caikit.interfaces.common.data_model: {candidate_type}"
             )
             continue
-        # ...And the containing library's data model
-        lib_base = sys.modules[module_class.__module__.partition(".")[0]]
-        if hasattr(lib_base, "data_model"):
-            candidate_type = _extract_type_from_pymodule(
-                lib_base.data_model, module_class, type_name
-            )
+        # ...And the data model within the interfaces, including those defined in library
+        try:
+            candidate_type = DataBase.get_class_for_name(type_name)
```

Comment:
The parameter `module_class` had only been passed around all the functions to get the library data model. We can instead get this info through with the recently exposed `get_class_for_name` function

Reply:
awesome!